tests/test_0692_fsspec.py

```diff
@@ -66,3 +67,32 @@ def test_open_fsspec_xrootd():
         data = f["Events/run"].array(library="np", entry_stop=20)
         assert len(data) == 20
         assert (data == 194778).all()
+
+
+@pytest.mark.network
+def test_fsspec_chunks():
+    # Use the local HTTP server to serve test.root
+    url = "https://github.com/scikit-hep/scikit-hep-testdata/raw/v0.4.33/src/skhep_testdata/data/uproot-issue121.root"
+
+    notifications = queue.Queue()
+    with uproot.source.fsspec.FSSpecSource(url) as source:
+        import time
+
+        start_time = time.time()
+        chunks = source.chunks(
+            [(0, 100), (50, 55), (200, 400)], notifications=notifications
+        )
+        time_to_get_chunks = time.time() - start_time
+        # TODO: this might not be the best way to test it is non-blocking
+        # This is non-blocking, so it should be fast (0.0005390644073486328 seconds last time I ran it)
+        assert (
+            time_to_get_chunks < 0.01
+        ), f"This should be fast (non-blocking). Elapsed: {time_to_get_chunks} seconds"
```

Comment:
No—do not test the blockingness. In CI, it will be running on some machine that might be inherently slower or something else might be running at the same time, and users with the sdist are invited to run the tests on their own machine—any of these can make the test fail for false reasons.

Basically, we can't test invisible aspects of computing. The performance characteristics are inside the black box, and tests can only test black-box outputs. But we can just be sure to code it in a way that has the right characteristics.